### PR TITLE
aichat: generated questions will silently fail

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/35929

Currently, the response from 'generate questions' doesn't handle the failure state. We could potentially show the same premium upsell UI with a 'retry' button, but the action here is redundant because you cannot 'retry' generated questions. This is a band-aid fix that will silently fail and reset, ensuring the 'generate question' button is not stuck in a loading state.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by aguscruiz** [comment]
@nullhook  this means that the user won't be able to send prompts either, right? And this could happen for both free and premium users?

Just to try and figure out the actions:

**Free user:**
Message: Couldn't generate suggested questions because you've reached the response limit. Switch to Premium to get a higher response limit and more advanced models, or try again in [time].

Actions:
- Go premium
- Dismiss? > This would mean the user is stuck until the rate limit resets

**Premium user:**
Message: You've reached the premium rate limit. The limit will be reset in [time]. In the meantime, you can switch to another model model to continue.

Action:
- Dismiss

@mattmcalister @bbondy Does this sound right?

File: components/ai_chat/core/browser/conversation_driver.cc [hunk 1]
```diff
@@ -583,17 +583,24 @@ void ConversationDriver::GenerateQuestions() {
...
 
-  suggestions_.insert(suggestions_.end(), result.begin(), result.end());
-  suggestion_generation_status_ =
-      mojom::SuggestionGenerationStatus::HasGenerated;
+  if (result.has_value()) {
+    suggestions_.insert(suggestions_.end(), result->begin(), result->end());
+    suggestion_generation_status_ =
+        mojom::SuggestionGenerationStatus::HasGenerated;
+  } else {
+    // TODO(nullhook): Set a specialized error state generated questions
```
Comment (petemill) [inline components/ai_chat/core/browser/conversation_driver.cc:599]: Please create a github issue for this and tag @aguscruiz so we can also get design for it

File: components/ai_chat/core/browser/engine/engine_consumer.h [hunk 2]
```diff
@@ -23,8 +23,8 @@ namespace ai_chat {
 // subclass) or remote (invoked via network requests).
 class EngineConsumer {
  public:
-  using SuggestedQuestionsCallback =
-      base::OnceCallback<void(std::vector<std::string>)>;
+  using SuggestedQuestionsCallback = base::OnceCallback<void(
+      base::expected<std::vector<std::string>, mojom::APIError>)>;
```
Comment (petemill) [inline components/ai_chat/core/browser/engine/engine_consumer.h:27]: Why don't we make a type (via `using`) for this like we do for the other things - e.g. `SuggestedQuestionsResult`?

Comment (nullhook) [reply components/ai_chat/core/browser/engine/engine_consumer.h]: done

**Review by petemill — APPROVED** [review]

File: components/ai_chat/core/browser/conversation_driver.cc (hunk 1 above)
Comment (nullhook) [reply components/ai_chat/core/browser/conversation_driver.cc:642]: https://github.com/brave/brave-browser/issues/36540

**Pull request merged**